File: Den.Dev.Orion/Den.Dev.Orion/Models/ApiContentType.cs
// <copyright file="APIContentType.cs" company="Den Delimarsky">
// Developed by Den Delimarsky.
// Den Delimarsky licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
// The underlying API powering Den.Dev.Orion is managed by Halo Studios and Microsoft. This wrapper is not endorsed by Halo Studios or Microsoft.
// </copyright>

namespace Den.Dev.Orion.Models
{
    /// <summary>
    /// Type of content to return for the Halo API.
    /// </summary>
    public enum APIContentType
    {
        /// <summary>
        /// JSON data format.
        /// </summary>
        [ContentType(HeaderValue = "application/json")]
        Json,

        /// <summary>
        /// <see href="https://microsoft.github.io/bond/manual/bond_cs.html#compact-binary">Microsoft Bond Compact Binary</see> data format.
        /// </summary>
        [ContentType(HeaderValue = "application/x-bond-compact-binary")]
        BondCompactBinary,
    }
}

File: src/node/src/models/common/api-content-type.ts
/**
 * API content types supported by the Halo API.
 *
 * Most endpoints use JSON, but some UGC-related endpoints
 * use Bond Compact Binary for binary asset data.
 */
export const ApiContentType = {
  /** Standard JSON content type */
  Json: 'json',
  /** Bond compact binary format for binary data */
  BondCompactBinary: 'bond',
} as const;

/**
 * Type representing valid API content types.
 */
export type ApiContentType = (typeof ApiContentType)[keyof typeof ApiContentType];

/**
 * Get the HTTP Content-Type header value for a content type.
 *
 * @param contentType - The content type enum value
 * @returns The corresponding Content-Type header value
 */
export function getContentTypeHeader(contentType: ApiContentType): string {
  switch (contentType) {
    case ApiContentType.Json:
      return 'application/json';
    case ApiContentType.BondCompactBinary:
      return 'application/x-bond-compact-binary';
    default:
      return 'application/json';
  }
}
